agents/loco_mc_agent.py

```diff
@@ -133,8 +131,8 @@ def send_text_command_to_agent(sid, command):
             logical_form = {}
             status = ""
             try:
-                chat_parse = self.chat_parser.get_logical_form(
-                    chat=command, parsing_model=self.chat_parser.parsing_model
+                chat_parse = self.perception_modules["language_understanding"].get_logical_form(
```

Comment:
why is this here?  shouldn't the chat be added, and then this is handled by the normal perceive() call after?

Reply:
yeah but why do we need to get_logical_form again?   like can't we get it from memory?

Reply:
Not sure I follow - in this case we just want to query the parser and return the logical form. What is the alternate route here without having to wait for the logical form to get written to memory ?

Reply:
lol I think we _should_ wait for it to get written to memory!  that way what gets sent to dashboard is definitely whatever the agent used/saw